rpcserver.go

```diff
@@ -3299,6 +3299,14 @@ func handleSearchRawTransactions(s *rpcServer, cmd interface{}, closeChan <-chan
 // handleSendRawTransaction implements the sendrawtransaction command.
 func handleSendRawTransaction(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
 	c := cmd.(*btcjson.SendRawTransactionCmd)
+	if s.cfg.TxIndex == nil {
```

Comment:
Seems like a pretty big downgrade if we now require the `txindex` just to _send_ a transaction over the interface. Instead, during validation in the mempool, we know the direct fee since it's stored as part of the mempool transaction entry. So we can put the logic deeper into the mempool, accepting an optional argument of the max fee that the resulting transaction should be allowed to have.